aea/helpers/profiling.py

```diff
@@ -196,57 +194,23 @@ def get_profile_data(self) -> Dict:
                 "amount": threading.active_count(),
                 "names": [i.name for i in threading.enumerate()],
             },
-            "objects_present": self.get_objects_instances(),
-            "objects_created": self.get_objects_created(),
-            "dialogues_by_type": get_dialogues_by_type(),
+            "objects_present": {k: v[0] - v[1] for k, v in self.object_counts.items()},
+            "objects_created": {k: v[0] for k, v in self.object_counts.items()},
+            "most_common_objects_in_gc": get_most_common_objects_in_gc(),
         }
 
-    def get_objects_instances(self) -> Dict:
-        """Return dict with counted object instances present now."""
-        result: Dict = Counter()
-
-        lock.acquire()
-        try:
-            for obj_type in self._objects_instances_to_count:
-                result[obj_type.__name__] += 0
-
-            for obj in gc.get_objects():
-                for obj_type in self._objects_instances_to_count:
-                    if isinstance(obj, obj_type):
-                        result[obj_type.__name__] += 1
-        finally:
-            lock.release()
-        return result
-
-    def get_objects_created(self) -> Dict:
-        """Return dict with counted object instances created."""
-        return self._counter
-
-
-def get_dialogues_by_type() -> Dict:
-    """Return dict with the number of dialogues by type."""
 
-    dialogues_by_type: Dict[str, int] = {"incomplete_to_complete_dialogue_labels": 0}
+def get_most_common_objects_in_gc(number: int = 10) -> List[Tuple[str, int]]:
+    """Get the highest-count objects in the garbage collector."""
 
+    object_count: CounterType = Counter()
     lock.acquire()
     try:
         for obj in gc.get_objects():
-            if isinstance(obj, Dialogue):
-
-                dialogue_type = str(type(obj).__name__)
-
-                if dialogue_type not in dialogues_by_type:
-                    dialogues_by_type[dialogue_type] = 1
-                else:
-                    dialogues_by_type[dialogue_type] += 1
-
-            elif isinstance(obj, BasicDialoguesStorage):
-                # Count incomplete dialogues by type
-                dialogues_by_type["incomplete_to_complete_dialogue_labels"] += len(
-                    list(
-                        obj._incomplete_to_complete_dialogue_labels.values()  # pylint: disable=protected-access
-                    )
-                )
+            # not all objects have the __class__ attribute
+            object_type = type(obj).__name__
+            if object_type not in PROFILER_TYPE_BLACK_LIST:
+                object_count[object_type] = object_count.get(object_type, 0) + 1
```

Comment:
I like this, but not needed when using `Counter` (only needed when using dict counters to avoid the if else). Can simply use `object_count[object_type] += 1` in this case